source/s3_client.c

```diff
@@ -304,10 +307,16 @@ PyObject *aws_py_s3_client_new(PyObject *self, PyObject *args) {
 
     struct aws_signing_config_aws *signing_config = NULL;
     struct aws_credentials *anonymous_credentials = NULL;
+    struct aws_byte_cursor *network_interface_names = NULL;
+    int num_network_interface_names = 0;
```

Comment:
```suggestion
    size_t num_network_interface_names = 0;
```
right? you assign to this later like `num_network_interface_names = (size_t)listSize;`